<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>728</width>
    <height>558</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Advent Of Code</string>
  </property>
  <widget class="QWidget" name="m_central_widget">
   <layout class="QVBoxLayout" name="m_central_widget_vertical_layout">
    <property name="spacing">
     <number>5</number>
    </property>
    <property name="leftMargin">
     <number>5</number>
    </property>
    <property name="topMargin">
     <number>5</number>
    </property>
    <property name="rightMargin">
     <number>5</number>
    </property>
    <property name="bottomMargin">
     <number>5</number>
    </property>
    <item>
     <widget class="QWidget" name="m_common_top_widget" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="MinimumExpanding" vsizetype="Maximum">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <layout class="QHBoxLayout" name="m_common_top_widget_horizontal_layout">
       <property name="spacing">
        <number>5</number>
       </property>
       <property name="leftMargin">
        <number>0</number>
       </property>
       <property name="topMargin">
        <number>0</number>
       </property>
       <property name="rightMargin">
        <number>0</number>
       </property>
       <property name="bottomMargin">
        <number>0</number>
       </property>
       <item>
        <widget class="QLabel" name="m_label_year">
         <property name="text">
          <string>Year</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QSpinBox" name="m_spin_box_year">
         <property name="minimum">
          <number>2015</number>
         </property>
         <property name="maximum">
          <number>2023</number>
         </property>
         <property name="value">
          <number>2023</number>
         </property>
        </widget>
       </item>
       <item>
        <spacer name="m_common_top_horizontal_spacer">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>40</width>
           <height>20</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="m_push_button_update_cookies">
         <property name="text">
          <string>Update cookies</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="m_push_button_sources">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Maximum" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Sources</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QTabWidget" name="m_tab_widget">
      <property name="sizePolicy">
       <sizepolicy hsizetype="MinimumExpanding" vsizetype="MinimumExpanding">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="m_solvers_tab">
       <attribute name="title">
        <string>Solvers</string>
       </attribute>
       <layout class="QVBoxLayout" name="m_solvers_tab_vertical_layout">
        <property name="spacing">
         <number>5</number>
        </property>
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item>
         <widget class="QWidget" name="m_solver_selection_widget" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <layout class="QHBoxLayout" name="m_solver_selection_widget_horizontal_layout">
           <property name="spacing">
            <number>5</number>
           </property>
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
            <widget class="QLabel" name="m_label_day">
             <property name="text">
              <string>Day</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="m_spin_box_day">
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>25</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="m_label_puzzle">
             <property name="text">
              <string>Puzzle</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QSpinBox" name="m_spin_box_puzzle">
             <property name="minimum">
              <number>1</number>
             </property>
             <property name="maximum">
              <number>2</number>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QCheckBox" name="m_check_box_use_last_input">
             <property name="text">
              <string>Use last input</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPushButton" name="m_push_button_solve">
             <property name="font">
              <font>
               <pointsize>15</pointsize>
              </font>
             </property>
             <property name="text">
              <string>SOLVE</string>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="m_horizontal_spacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>40</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QWidget" name="m_solver_output_widget" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Maximum">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <layout class="QVBoxLayout" name="m_solver_output_widget_vertical_layout">
           <property name="spacing">
            <number>5</number>
           </property>
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
            <widget class="QPushButton" name="m_push_button_solver_output">
             <property name="text">
              <string>Solver Output</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QPlainTextEdit" name="m_plain_text_edit_solver_output">
             <property name="maximumSize">
              <size>
               <width>16777215</width>
               <height>100</height>
              </size>
             </property>
             <property name="font">
              <font>
               <family>Ubuntu Mono</family>
              </font>
             </property>
             <property name="readOnly">
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QWidget" name="m_program_io_widget" native="true">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="MinimumExpanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <layout class="QHBoxLayout" name="m_program_io_widget_horizontal_layout">
           <property name="spacing">
            <number>5</number>
           </property>
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
            <widget class="QWidget" name="m_program_input_widget" native="true">
             <layout class="QVBoxLayout" name="m_program_input_widget_vertical_layout">
              <property name="spacing">
               <number>5</number>
              </property>
              <property name="leftMargin">
               <number>0</number>
              </property>
              <property name="topMargin">
               <number>0</number>
              </property>
              <property name="rightMargin">
               <number>0</number>
              </property>
              <property name="bottomMargin">
               <number>0</number>
              </property>
              <item>
               <widget class="QPushButton" name="m_push_button_input">
                <property name="text">
                 <string>Program Input</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPlainTextEdit" name="m_plain_text_edit_input">
                <property name="font">
                 <font>
                  <family>Ubuntu Mono</family>
                 </font>
                </property>
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
           <item>
            <widget class="QWidget" name="m_program_output_widget" native="true">
             <layout class="QVBoxLayout" name="m_program_output_widget_vertical_layout">
              <property name="spacing">
               <number>5</number>
              </property>
              <property name="leftMargin">
               <number>0</number>
              </property>
              <property name="topMargin">
               <number>0</number>
              </property>
              <property name="rightMargin">
               <number>0</number>
              </property>
              <property name="bottomMargin">
               <number>0</number>
              </property>
              <item>
               <widget class="QPushButton" name="m_push_button_program_output">
                <property name="text">
                 <string>Program Output</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPlainTextEdit" name="m_plain_text_edit_program_output">
                <property name="sizePolicy">
                 <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
                  <horstretch>0</horstretch>
                  <verstretch>0</verstretch>
                 </sizepolicy>
                </property>
                <property name="maximumSize">
                 <size>
                  <width>16777215</width>
                  <height>16777215</height>
                 </size>
                </property>
                <property name="font">
                 <font>
                  <family>Ubuntu Mono</family>
                 </font>
                </property>
                <property name="readOnly">
                 <bool>true</bool>
                </property>
               </widget>
              </item>
             </layout>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="m_leader_board_tabs">
       <attribute name="title">
        <string>Leaderboards</string>
       </attribute>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
